File: ITSE-2457/Module 7/Car Lot/src/Car.java
/*
 * ITSE-2457; OOP Java; M/W 2pm
 * Written by Andrew Pettus
 * Apr 2 2023
 * Module 7 Lab 1
 * Car Lot Inventory
 */

public class Car {
    // variables
    private String make;
    private String model;
    private int year;
    private double price;

    //object defintion
    public Car(String make, String model, int year, double price) {
        this.make = make;
        this.model = model;
        this.year = year;
        this.price = price;
    }
    // gettersetters
    public String getMake() {
        return make;
    }
    public void setMake(String make) {
        this.make = make;
    }
    public String getModel() {
        return model;
    }
    public void setModel(String model) {
        this.model = model;
    }
    public int getYear() {
        return year;
    }
    public void setYear(int year) {
        this.year = year;
    }
    public double getPrice() {
        return price;
    }
    public void setPrice(double price) {
        this.price = price;
    }
    // overridden toString
    public String toString() {
        return make + " " + model + " " + year + " $" + price;
    }
}

File: ITSE-2445-Programming 3 - Data Structures - C++/Module 6 Classes and Objects/CarDemo/Car.cpp
//  Car.cpp
//  Car Class implementation
//  Created by Kevin Roark on 3/26/23
#include "Car.h"
#include <string>
#include <iomanip>
using namespace std;
/*
 Getters (Assessors)
 */
string Car::getMake() const
{
    return make;
}
string Car::getModel() const
{
    return model;
}
int Car::getYear() const
{
    return year;
}
double Car::getPrice() const
{
    return price;
}

/*
 setters (Mutators)
 */
void Car::setMake(string pMake)
{
    make = pMake;
}
void Car::setModel(string pModel)
{
    model = pModel;
}
void Car::setYear(int pYear)
{
    year = pYear;
}
void Car::setPrice(double pPrice)
{
    price = pPrice;
}

/*
 Constructors - with Example of an Overloaded Constructor
 */
Car::Car(string pMake, string pModel, int pYear, double pPrice)
{
    make = pMake;
    model = pModel;
    year = pYear;
    price = pPrice;
}

Car::Car(string pMake, string pModel, int pYear)
{
    make = pMake;
    model = pModel;
    year = pYear;
    price = 0.0;
}
Car::Car()
{
    make = "Unknown Make";
    model = "Unknown Model";
    year = 2023;
    price = 0.0;
}
//functions
void Car::Print()
{
    cout << "Make: " << make << endl;
    cout << "Model: " << model << endl;
    cout << "Year: " << year << endl;
    cout << showpoint << fixed << setprecision(2);
    cout << "Price: $" << price << endl;
}
